src/Features/Core/Diagnostics/EngineV1/DiagnosticAnalyzerDriver.cs

```diff
@@ -259,12 +254,19 @@ private ImmutableArray<SyntaxNode> GetSyntaxNodesToAnalyze()
                     }
                     catch (Exception e) when (CatchAnalyzerException(e, analyzer))
                     {
-                        var exceptionDiagnostics = AnalyzerExceptionToDiagnostics(analyzer, e, _cancellationToken);
-                        return GetFilteredDocumentDiagnostics(exceptionDiagnostics, compilation).ToImmutableArray();
+                        var exceptionDiagnostic = AnalyzerExceptionToDiagnostic(analyzer, e, _cancellationToken);
+                        if (exceptionDiagnostic != null)
+                        {
+                            ReportAnalyzerExceptionDiagnostic(analyzer, exceptionDiagnostic, compilation);
+                        }
+
+                        return ImmutableArray<Diagnostic>.Empty;
                     }
                 }
 
-                var analyzerActions = await this.GetAnalyzerActionsAsync(analyzer, diagnostics.Add).ConfigureAwait(false);
+                var handler = WorkspaceAnalyzerManager.RegisterAnalyzerExceptionDiagnosticHandler(analyzer, this.Project);
```

Comment:
I don't understand the full context of your question. Exception from analyzer? Once the handler is registered, it ends up wrapping the exception diagnostic event with workspace/project info and passes it to AbstractHostDiagnosticUpdateSource.